### PR TITLE
HDFS is unsupported

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

<details>
    <summary>Modify your CI run</summary>

**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  All with Azure
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

</details>

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit ea303e0d6346cd1bdd96c87ef767856db9425133 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64126/ea303e0d6346cd1bdd96c87ef767856db9425133/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: docs/en/operations/storing-data.md [hunk 1]
```diff
@@ -464,9 +466,11 @@ In this sample configuration:
 </clickhouse>
 ```
 
+Keep in mind that HDFS does not work at all.
```
Comment (kssenii) [inline docs/en/operations/storing-data.md:469]: May be this message is too much? "At all" is not really true... If if did not work at all, there would have been no sense having hdfs in documentation at all.

File: docs/en/operations/storing-data.md [hunk 2]
```diff
@@ -428,12 +428,14 @@ Examples of working configurations can be found in integration tests directory (
...
 
-## Using HDFS storage {#hdfs-storage}
+## Using HDFS storage (Unsupported)
 
 In this sample configuration:
-- the disk is of type `hdfs`
+- the disk is of type `hdfs` (unsupported)
 - the data is hosted at `hdfs://hdfs1:9000/clickhouse/`
 
+By the way, HDFS is unsupported and won't work.
```
Comment (kssenii) [inline docs/en/operations/storing-data.md:437]:
```suggestion
By the way, HDFS is unsupported and therefore there might be issues when using it. Feel free to make a pull request with the fix if the issues arises.
```

File: docs/en/operations/storing-data.md (hunk 1 above)
Comment (alexey-milovidov) [inline docs/en/operations/storing-data.md:469]:
```suggestion
Keep in mind that HDFS may not work in corner cases.
```

**Review by kssenii — APPROVED** [review]

File: docs/en/operations/storing-data.md (hunk 1 above)
Comment (den-crane) [reply docs/en/operations/storing-data.md]: @kssenii latest HDFS have a hard limit 1million files per folder. 
Clickhouse stores all objects in the the single HDFS folder (for HDFS as external storage, unlike S3).
The average table with 500 columns reaches this limit 1million objects easily, it's less 1000 active+inactive wide parts, after that a table is unable to insert or to merge.

https://github.com/ClickHouse/ClickHouse/issues/43112

**Pull request merged**